src/posthog-core.ts

```diff
@@ -744,8 +744,19 @@ export class PostHog {
         }
 
         if (event_name === '$identify') {
-            data['$set'] = options?.['$set']
-            data['$set_once'] = options?.['$set_once']
+            const setProperties = options?.['$set']
+            if (setProperties) {
+                data['$set'] = options?.['$set']
+            }
+            const setOnceProperties = this._calculate_set_once_properties(options?.['$set_once'])
+            if (setOnceProperties) {
+                data['$set_once'] = setOnceProperties
+            }
+        } else {
+            const setOnceProperties = this._calculate_set_once_properties()
+            if (setOnceProperties) {
+                data['$set_once'] = this._calculate_set_once_properties()
```

Comment:
yep, coming to this a second time it feels like it'd be more clear to handle $set and $set_once separately

but I've never dealt with them previously so it might be obvious if you know what they do 🙈

Reply:
I've tidied this up, you're right it doesn't need a separate branch for identify events, as only identify events would be passing `options?.['$set_once']` anyway